pkg/storegateway/bucket_index_metadata_fetcher.go

```diff
@@ -49,37 +52,37 @@ func NewBucketIndexMetadataFetcher(
 		logger:      logger,
 		filters:     filters,
 		modifiers:   modifiers,
-		metrics:     newFetcherMetrics(reg),
+		metrics:     block.NewFetcherMetrics(reg, [][]string{{corruptedBucketIndex}, {noBucketIndex}}, nil),
 	}
 }
 
 // Fetch implements metadata.MetadataFetcher.
 func (f *BucketIndexMetadataFetcher) Fetch(ctx context.Context) (metas map[ulid.ULID]*metadata.Meta, partial map[ulid.ULID]error, err error) {
-	f.metrics.resetTx()
+	f.metrics.ResetTx()
```

Comment:
**thought(non-blocking):** The [upstream ResetTx function](https://github.com/thanos-io/thanos/blob/8cf129b1451d9cad160dd440bd8064a89b49d96d/pkg/extprom/tx_gauge.go#L47) is not goroutine safe. Would it make sense to propagate that warning here with a comment indicating the same?

Reply:
The whole `MetadataFetcher.Fetch()` is not required to be goroutine-safe (in Thanos it's not implemented in a goroutine-safe way too). I've updated the function comment.